### PR TITLE
[Routing] Move to GenericHost

### PR DESCRIPTION
**Pull request by Kahbazi**
Part of #20964

### CODE REVIEW

**Review by javiercn — APPROVED** [review]
Thanks for the contribution!

Waiting for the tests to pass to merge

File: src/Http/Routing/test/FunctionalTests/EndpointRoutingIntegrationTest.cs [hunk 1]
```diff
@@ -31,20 +32,26 @@ public class EndpointRoutingIntegrationTest
...
+                        .UseTestServer();
+                })
+                .ConfigureServices(services =>
+                {
+                    services.AddAuthorization();
+                    services.AddRouting();
+                })
+                .Build();
+
+            using var server = host.GetTestServer();
```
Comment (BrennanConroy) [inline src/Http/Routing/test/FunctionalTests/EndpointRoutingIntegrationTest.cs:54]: Looks like you forgot to start the app. `await host.StartAsync();`

File: src/Http/Routing/test/FunctionalTests/EndpointRoutingIntegrationTest.cs [hunk 2]
```diff
@@ -31,20 +32,26 @@ public class EndpointRoutingIntegrationTest
...
-                app.UseEndpoints(b => b.Map("/", TestDelegate));
-            })
-           .ConfigureServices(services =>
-           {
-               services.AddAuthorization();
-               services.AddRouting();
-           });
-
-            using var server = new TestServer(builder);
+            var host = new HostBuilder()
```
Comment (BrennanConroy) [inline src/Http/Routing/test/FunctionalTests/EndpointRoutingIntegrationTest.cs:35]:
```suggestion
            using var host = new HostBuilder()
```

**Comment by Tratcher** [comment]
I found a few more:
https://github.com/dotnet/aspnetcore/blob/527096a3415b1b3f7d1f00098255d4cede9160b4/src/Http/Routing/test/FunctionalTests/EndpointRoutingSampleTest.cs#L21
https://github.com/dotnet/aspnetcore/blob/527096a3415b1b3f7d1f00098255d4cede9160b4/src/Http/Routing/test/testassets/RoutingWebSite/Program.cs#L60
https://github.com/dotnet/aspnetcore/blob/527096a3415b1b3f7d1f00098255d4cede9160b4/src/Http/Routing/test/testassets/RoutingSandbox/Program.cs#L60
https://github.com/dotnet/aspnetcore/blob/527096a3415b1b3f7d1f00098255d4cede9160b4/src/Http/Routing/test/testassets/Benchmarks/Program.cs#L17
https://github.com/dotnet/aspnetcore/blob/527096a3415b1b3f7d1f00098255d4cede9160b4/src/Http/Routing/test/FunctionalTests/Benchmarks/EndpointRoutingBenchmarkTest.cs#L22
https://github.com/dotnet/aspnetcore/blob/527096a3415b1b3f7d1f00098255d4cede9160b4/src/Http/Routing/test/FunctionalTests/WebHostBuilderExtensionsTest.cs#L82

File: src/Http/Routing/test/FunctionalTests/EndpointRoutingIntegrationTest.cs (hunk 1 above)
Comment (BrennanConroy) [reply src/Http/Routing/test/FunctionalTests/EndpointRoutingIntegrationTest.cs:54]: I wont comment on the rest of the places as they look like they have the same issue

**Comment by Tratcher** [comment]
There's an unrelated CI build break. We'll look into it.

**Review by JamesNK — APPROVED** [review]
Looks good. Test failure seems unrelated. I've triggered a rerun.

**Comment by davidfowl** [comment]
Why aren't we using auto merge on these?

**Review by BrennanConroy — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Thanks for all these @Kahbazi !

**Pull request merged**